llama_index/llms/llama_cpp.py

```diff
@@ -144,7 +149,7 @@ def class_name(cls) -> str:
     def metadata(self) -> LLMMetadata:
         """LLM metadata."""
         return LLMMetadata(
-            context_window=self.context_window,
+            context_window=self._model.params.n_ctx,
```

Comment:
This removes the statefulness, where this class houses `self.context_window` but the real one that matters is within `self._model`